src/include/function/list/functions/list_product_function.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include "common/vector/value_vector.h"
+
+namespace kuzu {
+namespace function {
+
+struct ListProduct {
+    template<typename T>
+    static inline void operation(common::list_entry_t& input, T& result,
+        common::ValueVector& inputVector, common::ValueVector& resultVector) {
+        auto inputValues =
+            reinterpret_cast<T*>(common::ListVector::getListValues(&inputVector, input));
+        auto inputDataVector = common::ListVector::getDataVector(&inputVector);
+        result = 1;
+        for (auto i = 0; i < input.size; i++) {
+            if (inputDataVector->isNull(input.offset + i)) {
+                continue;
+            }
+            result *= inputValues[i];
```

Comment:
This is probably fine. But we are adding new interfaces to avoid direct access of data inside vector. So you should probably do

```
for (auto i = 0; i < input.size; i++) {
       if (inputDataVector->isNull(input.offset + i)) {
                 continue;
         }
          result *= inputDataVector.getValue<T>(input.offset + i);
```

Apply the same change to list_sum